test/e2e/specs/interactivity/tovdom-islands.spec.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * Internal dependencies
+ */
+import { test, expect } from './fixtures';
+
+test.describe( 'toVdom - islands', () => {
+	test.beforeAll( async ( { interactivityUtils: utils } ) => {
+		await utils.activatePlugins();
+		await utils.addPostWithBlock( 'test/tovdom-islands' );
+	} );
+	test.beforeEach( async ( { interactivityUtils: utils, page } ) => {
+		await page.goto( utils.getLink( 'test/tovdom-islands' ) );
+	} );
+	test.afterAll( async ( { interactivityUtils: utils } ) => {
+		await utils.deactivatePlugins();
+		await utils.deleteAllPosts();
+	} );
+
+	test( 'directives that are not inside islands should not be hydrated', async ( {
+		page,
+	} ) => {
+		const el = page.getByTestId( 'not inside an island' );
+		await expect( el ).toBeVisible();
+	} );
+
+	test( 'directives that are inside islands should be hydrated', async ( {
+		page,
+	} ) => {
+		const el = page.getByTestId( 'inside an island' );
+		await expect( el ).toBeHidden();
+	} );
+
+	test( 'directives that are inside inner blocks of isolated islands should not be hydrated', async ( {
+		page,
+	} ) => {
+		const el = page.getByTestId(
+			'inside an inner block of an isolated island'
+		);
+		await expect( el ).toBeVisible();
+	} );
+
+	test( 'directives inside islands should not be hydrated twice', async ( {
```

Comment:
This test became flaky - #58332.

Reply:
I think @DAreRodz was already working on a fix, weren't you David?

Reply:
Yup, fixed in https://github.com/WordPress/gutenberg/pull/58377/commits/f45282b9ab7c01e58e67705ab573d1087c482ebe (part of https://github.com/WordPress/gutenberg/pull/58377, which is ready for review).